File: FlutterPlugin/android/src/main/kotlin/so/daro/flutter/DaroAdInstanceManager.kt
package so.daro.flutter

import io.flutter.plugin.platform.PlatformView

object DaroAdInstanceManager {
    private val ads = mutableMapOf<Int, PlatformView>()

    fun store(ad: PlatformView, adId: Int) {
        ads[adId] = ad
    }

    fun get(adId: Int): PlatformView? {
        return ads[adId]
    }

    fun remove(adId: Int) {
        val ad = ads.remove(adId)
        ad?.dispose()
    }
}

File: FlutterPlugin/ios/Classes/DaroAdInstanceManager.swift
import Flutter

class DaroAdInstanceManager {
    static let shared = DaroAdInstanceManager()

    private var ads: [Int: FlutterPlatformView] = [:]

    func store(_ ad: FlutterPlatformView, forId adId: Int) {
        ads[adId] = ad
    }

    func get(forId adId: Int) -> FlutterPlatformView? {
        return ads[adId]
    }

    func remove(forId adId: Int) {
        ads.removeValue(forKey: adId)
    }
}
